mitodl/ocw-studio
# Cc/fix pagination

#### Pre-Flight checklist


#### What are the relevant tickets?
#1209 

#### What's this PR do?
This refactors pagination for pages that are using the `useURLParamFilter` and fixes an issue where pagination was not preserving query parameters.

This also includes some tinkering with our types for react-router / history. See https://github.com/mitodl/ocw-studio/pull/1226/commits/ac6e7ff27399f268e12d90118f7aec83c0c59ace

#### How should this be manually tested?
On master:
1. On master...
2. Find any content listing that contains lots of item. For example http://yourname.ocw-studio.odl.local:8043/sites/8-591j-systems-biology-fall-2004/type/resource
3. Add a search query in the filter box such that you still have more content than fits on one page. For example, "s"
4. Try to go to "next page" with navigation at bottom. It'll work for a moment, then you'll be brought back to the first page. 😠 
5. Repeat on this branch, `cc/fix-pagination`
    - pagination should work and preserve query parameters
    - pagination should work for content listing and at /sites
